### DISCUSSION TITLE
Handling Realtime channel disconnection error

### DISCUSSION THREAD

**Discussion by AkmenZ** [supabase/supabase#16422] [Questions]
Hello. 
Im trying to implement chat feature in my flutter application. The chat works fine until the app is minimised/idle for a while. 

The error stack shows:
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception:
#0      SupabaseStreamBuilder._addException
supabase_stream_builder.dart:442
#1      SupabaseStreamBuilder._getStreamData.<anonymous closure>
supabase_stream_builder.dart:348
#2      RealtimeChannel.subscribe.<anonymous closure>
realtime_channel.dart:202
#3      RealtimeChannel.onError.<anonymous closure>
realtime_channel.dart:331
#4      RealtimeChannel.trigger
realtime_channel.dart:569
#5      RealtimeClient._triggerChanError
realtime_client.dart:408
#6      RealtimeClient._onConnClose
realtime_client.dart:389
#7      RealtimeClient.connect.<anonymous closure>
realtime_client.dart:147
#8      _RootZone.runGuarded (dart:async/zone.dart:1582:10)
#9      _BufferingStreamSubscription._sendDone.sendD<…>

How i understand it is that it closes the channel, throws error and never resumes it. I'm using Riverpod for state management with StreamProvider that should automatically handle the state. 
Anyone else has faced a similar issue?

Thanks

![Simulator Screen Shot - iPhone 14 Pro Max - 2023-08-10 at 11 02 41](https://github.com/supabase/supabase/assets/78148307/52583874-62f9-4603-a7a0-f4970caa4b7f)

**Comment by GaryAustin1** [comment]
Edit:   Not a Flutter answer, but maybe related on power down/idle.  I jumped on idle and disconnect and missed you are using Flutter.

Please see 
https://github.com/GaryAustin1/Realtime2
and
https://github.com/orgs/supabase/discussions/5641

One of the issues: https://github.com/supabase/realtime-js/issues/121
